### PR TITLE
Update Composer libraries to current versions

### PR DESCRIPTION
**Pull request by mbabker**
### Summary of Changes

Composer libraries are updated to current versions, the diff here includes all the production dependencies but PHPUnit is updated as well.  This is run using Composer 1.2.4 which accounts for some of the changes in the generated autoloader files.

### Testing Instructions

Joomla functions as normal.

### Documentation Changes Required

N/A

### CODE REVIEW

**Comment by zero-24** [comment]
Anything special to test? Especially the changes in the joomla vendor folder? The normal things looks good from here :smile:

**Comment by fastslack** [comment]
@test composer update works as expected. Joomla! CMS seems to works fine too.

File: libraries/vendor/joomla/session/Joomla/Session/Session.php [hunk 1]
```diff
@@ -649,7 +649,7 @@ public function destroy()
 		 */
 		if (isset($_COOKIE[session_name()]))
 		{
-			setcookie(session_name(), '', time() - 42000, $this->cookie_path, $this->cookie_domain);
+			$this->input->cookie(session_name(), '', 1);
```
Comment (zero-24) [inline libraries/vendor/joomla/session/Joomla/Session/Session.php]: this is an isssue @wilsonge found upstream so we need to wait for the fix?

**Comment by mbabker** [comment]
Updated.

**Comment by zero-24** [comment]
I have tested this item :white_check_mark: successfully on 42475c4d191b573f9b6702648c8c0f89fcf494e1<br /><br />:+1: I have not found any issue by navigation in backend and frontend as well as email sending still works (test mail)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13321">issues.joomla.org/tracker/joomla-cms/13321</a>.</sub>

**Pull request merged**